orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/instance/WaitForUpInstancesTask.groovy

```diff
@@ -154,4 +167,38 @@ class WaitForUpInstancesTask extends AbstractWaitingForInstancesTask {
     // no health indicators is indicative of being down
     return !healths || healths.any { it.state == 'Down' || it.state == 'OutOfService' }
   }
+
+  private static Map<String, Integer> getServerGroupCapacity(Stage stage, Map serverGroup) {
```

Comment:
Will get some proper unit tests of various edge-cases in here ... wanted to get feedback on approach first.